### ISSUE TITLE
[Bug]: Entity using native unit of measurement 'None' which is not a valid unit for the device class

### ISSUE THREAD

**Issue by alecsmro** [wlcrs/huawei_solar#245]
### Describe the issue

Even everything seems to work properly (except a bit longer delayed sensors values) i've got this persistent warning in log:

Entity sensor.inverter_power_factor (<class 'custom_components.huawei_solar.sensor.HuaweiSolarSensorEntity'>) is using native unit of measurement 'None' which is not a valid unit for the device class ('power_factor') it is using; Please update your configuration if your entity is manually configured, otherwise report it to the custom integration author.

Entity sensor.power_meter_power_factor (<class 'custom_components.huawei_solar.sensor.HuaweiSolarSensorEntity'>) is using native unit of measurement 'None' which is not a valid unit for the device class ('power_factor') it is using; Please update your configuration if your entity is manually configured, otherwise report it to the custom integration author.

Entity sensor.power_meter_reactive_power_2 (<class 'custom_components.huawei_solar.sensor.HuaweiSolarSensorEntity'>) is using native unit of measurement 'kVarh' which is not a valid unit for the device class ('reactive_power') it is using; Please update your configuration if your entity is manually configured, otherwise report it to the custom integration author.

### Bescribe your Huawei Solar Setup

Inverter Type: SUN2000-10KTL-M1
Inverter Firmware version: V100R001C00SPC148
SDongle present: yes  / Firmware 
Power meter present: three phase
Battery: None
Battery Firmware version: NONE

Huawei Solar version 1.2.0b7 -  also tried 1.2.0b5

### How do you connect to the inverter?

Via the SDongle, wireless connection

### Upload your Diagnostics File

{
  "home_assistant": {
    "installation_type": "Home Assistant OS",
    "version": "2023.1.1",
    "dev": false,
    "hassio": true,
    "virtualenv": false,
    "python_version": "3.10.7",
    "docker": true,
    "arch": "aarch64",
    "timezone": "Europe/Bucharest",
    "os_name": "Linux",
    "os_version": "5.15.76-v8",
    "supervisor": "2022.12.1",
    "host_os": "Home Assistant OS 9.4",
    "docker_version": "20.10.19",
    "chassis": "embedded",
    "run_as_root": true
  },
  "custom_components": {
    "sonoff": {
      "version": "3.3.1",
      "requirements": [
        "pycryptodome>=3.6.6"
      ]
    },
    "hacs": {
      "version": "1.29.0",
      "requirements": [
        "aiogithubapi>=22.10.1"
      ]
    },
    "fullykiosk": {
      "version": "1.1.0",
      "requirements": [
        "python-fullykiosk==0.0.11"
      ]
    },
    "tapo_control": {
      "version": "4.2.1",
      "requirements": [
        "pytapo==2.8",
        "onvif-zeep-async==1.2.0"
      ]
    },
    "huawei_solar": {
      "version": "1.2.0b5",
      "requirements": [
        "huawei-solar==2.2.0b6"
      ]
    },
    "midea_ac_lan": {
      "version": "v0.3.16-Beta7",
      "requirements": []
    },
    "xiaomi_cloud_map_extractor": {
      "version": "v2.2.0",
      "requirements": [
        "pillow",
        "pybase64",
        "python-miio",
        "requests",
        "pycryptodome"
      ]
    }
  },
  "integration_manifest": {
    "domain": "huawei_solar",
    "name": "Huawei Solar",
    "config_flow": true,
    "documentation": "https://www.home-assistant.io/integrations/huawei_solar",
    "issue_tracker": "https://github.com/wlcrs/huawei_solar/issues",
    "requirements": [
      "huawei-solar==2.2.0b6"
    ],
    "codeowners": [
      "@wlcrs"
    ],
    "iot_class": "local_polling",
    "version": "1.2.0b5",
    "loggers": [
      "huawei_solar",
      "pymodbus"
    ],
    "is_built_in": false
  },
  "data": {
    "config_entry_data": {
      "host": "192.168.1.21",
      "port": 502,
      "slave_ids": [
        1
      ],
      "enable_parameter_configuration": true,
      "username": null,
      "password": null
    },
    "slave_0": {
      "model_name": "SUN2000-10KTL-M1\u0000\u0000\u0000\u0000SUN2000-12",
      "pv_string_count": 2,
      "has_optimizers": 0,
      "battery_type": 0,
      "battery_1_type": 0,
      "battery_2_type": 0,
      "power_meter_type": 1,
      "supports_capacity_control": false
    },
    "slave_0_data": {
      "input_power": [
        0,
        "W"
      ],
      "line_voltage_A_B": [
        0.0,
        "V"
      ],
      "line_voltage_B_C": [
        0.0,
        "V"
      ],
      "line_voltage_C_A": [
        0.0,
        "V"
      ],
      "phase_A_voltage": [
        0.0,
        "V"
      ],
      "phase_B_voltage": [
        0.0,
        "V"
      ],
      "phase_C_voltage": [
        0.0,
        "V"
      ],
      "phase_A_current": [
        0.0,
        "A"
      ],
      "phase_B_current": [
        0.0,
        "A"
      ],
      "phase_C_current": [
        0.0,
        "A"
      ],
      "day_active_power_peak": [
        5593,
        "W"
      ],
      "active_power": [
        0,
        "W"
      ],
      "reactive_power": [
        0,
        "VA"
      ],
      "power_factor": [
        0.0,
        null
      ],
      "grid_frequency": [
        0.0,
        "Hz"
      ],
      "efficiency": [
        null,
        "%"
      ],
      "internal_temperature": [
        0.0,
        "\u00b0C"
      ],
      "insulation_resistance": [
        30.0,
        "MOhm"
      ],
      "device_status": [
        "Standby: no irradiation",
        null
      ],
      "fault_code": [
        0,
        null
      ],
      "startup_time": [
        "2023-01-07T06:01:00+00:00",
        null
      ],
      "shutdown_time": [
        "2023-01-07T14:17:12+00:00",
        null
      ],
      "accumulated_yield_energy": [
        1249.57,
        "kWh"
      ],
      "daily_yield_energy": [
        4.63,
        "kWh"
      ],
      "state_1": [
        [
          "Standby"
        ],
        null
      ],
      "state_2": [
        [
          "Locked",
          "PV disconnected",
          "DSP data collection"
        ],
        null
      ],
      "state_3": [
        [
          "On-grid",
          "Off-grid switch disabled"
        ],
        null
      ],
      "alarm_1": [
        [],
        null
      ],
      "alarm_2": [
        [],
        null
      ],
      "alarm_3": [
        [],
        null
      ],
      "pv_01_voltage": [
        0.0,
        "V"
      ],
      "pv_01_current": [
        0.0,
        "A"
      ],
      "pv_02_voltage": [
        0.0,
        "V"
      ],
      "pv_02_current": [
        0.0,
        "A"
      ],
      "grid_A_voltage": [
        233.1,
        "V"
      ],
      "grid_B_voltage": [
        233.3,
        "V"
      ],
      "grid_C_voltage": [
        234.6,
        "V"
      ],
      "active_grid_A_current": [
        -1.39,
        "I"
      ],
      "active_grid_B_current": [
        -0.41,
        "I"
      ],
      "active_grid_C_current": [
        -0.54,
        "I"
      ],
      "power_meter_active_power": [
        -231,
        "W"
      ],
      "power_meter_reactive_power": [
        433,
        "Var"
      ],
      "active_grid_power_factor": [
        -0.421,
        null
      ],
      "active_grid_frequency": [
        49.97,
        "Hz"
      ],
      "grid_exported_energy": [
        865.79,
        "kWh"
      ],
      "grid_accumulated_energy": [
        1835.03,
        "kWh"
      ],
      "grid_accumulated_reactive_power": [
        0.0,
        "kVarh"
      ],
      "meter_type": [
        1,
        null
      ],
      "active_grid_A_B_voltage": [
        403.8,
        "V"
      ],
      "active_grid_B_C_voltage": [
        405.2,
        "V"
      ],
      "active_grid_C_A_voltage": [
        405.0,
        "V"
      ],
      "active_grid_A_power": [
        -154,
        "W"
      ],
      "active_grid_B_power": [
        -15,
        "W"
      ],
      "active_grid_C_power": [
        -61,
        "W"
      ]
    },
    "slave_0_config_data": {}
  }
}

### Upload your relevant debug logs

```text
Entity sensor.inverter_power_factor (<class 'custom_components.huawei_solar.sensor.HuaweiSolarSensorEntity'>) is using native unit of measurement 'None' which is not a valid unit for the device class ('power_factor') it is using; Please update your configuration if your entity is manually configured, otherwise report it to the custom integration author.

Entity sensor.power_meter_power_factor (<class 'custom_components.huawei_solar.sensor.HuaweiSolarSensorEntity'>) is using native unit of measurement 'None' which is not a valid unit for the device class ('power_factor') it is using; Please update your configuration if your entity is manually configured, otherwise report it to the custom integration author.

Entity sensor.power_meter_reactive_power_2 (<class 'custom_components.huawei_solar.sensor.HuaweiSolarSensorEntity'>) is using native unit of measurement 'kVarh' which is not a valid unit for the device class ('reactive_power') it is using; Please update your configuration if your entity is manually configured, otherwise report it to the custom integration author.
```

### Please confirm the following:

- [ ] The problem is still present in the latest release of this integration.
- [ ] I did not find an existing issue describing this problem.
- [ ] I did upload the diagnostics-file that I could retrieve from  the 'Devices & Services Page'
- [ ] I increased the logging configuration, restarted HA, and have copied all relevant log lines from the 'Full Logs' into the textarea.

**Comment by tarmor1** [comment]
Connected to https://github.com/home-assistant/core/pull/85287 and fixed in HA [2023.1.2](https://github.com/home-assistant/core/releases/tag/2023.1.2)

**Comment by alecsmro** [comment]
@tarmor1 now it's fixed, like you said, but there is another one with a similar issue:
Entity sensor.power_meter_reactive_power_2 (<class 'custom_components.huawei_solar.sensor.HuaweiSolarSensorEntity'>) is using native unit of measurement 'kVarh' which is not a valid unit for the device class ('reactive_power') it is using; Please update your configuration if your entity is manually configured, otherwise report it to the custom integration author.

**Comment by wlcrs** [comment]
Fixed in 1.2.0
